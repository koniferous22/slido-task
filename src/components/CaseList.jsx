import React from 'react'

const CaseList = props => {
	return (
		<section id={props.id}>
			<h4 className="case-list-title">Past Events</h4>
			<ul className="case-list">
			{
				props.caseEntries.map((case_, index) => (
					<li key={index} onClick={() => props.handleSelectCase(index + (props.offset || 0))}>
						<i>{'[' + case_.date.toLocaleDateString() + '] '}</i>
						{case_.title.substring(0,50) + (case_.title.length > 50 ? '...' : '' )}
					</li>		
				))
			}
			</ul>
		</section>
	)	
}

export default CaseList
