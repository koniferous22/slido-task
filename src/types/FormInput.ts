export interface FormInputProps {
	name: string,
	label: string,
	type: string,
	value: string,
	errorFlag: boolean,
	onBlur: (s: string) => void,
	onChange: (s: string, e: React.ChangeEvent<HTMLInputElement>) => void
}
