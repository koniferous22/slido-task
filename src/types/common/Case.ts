export interface Case {
	title: string;
	description: string;
	date: Date;
}