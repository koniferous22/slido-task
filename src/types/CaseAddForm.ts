import { Case } from 'types/common/Case'
import { FormRecord, FormMaybeRecord } from 'types/form/FormRecord'

export interface CaseAddFormState {
	errorFlags: FormMaybeRecord<boolean>;
	values: FormRecord<string>;
}

export interface CaseAddFormProps {
	onSubmit: (case_: Case) => void;
}