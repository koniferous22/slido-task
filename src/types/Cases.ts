import { Case } from 'types/common/Case'

export interface CasesProps {
	caseEntries: Array<Case>;
	handleSelectCase: (_: number) => void;
	selectedCaseIndex: number;
}
