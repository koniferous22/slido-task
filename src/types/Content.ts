import { Case } from 'types/common/Case'

export interface ContentProps {
	displayedCase: Case | undefined;
	placeholder: string;
}
