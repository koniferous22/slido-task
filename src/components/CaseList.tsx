import React from 'react'

import { Case } from '../types/Case'

interface CaseListProps {
	id: string,
	caseEntries: Array<Case>
}

const CaseList = (props: any) => {
	return (
		<section id={props.id}>
			<h4 className="case-list-title">Past Events</h4>
			<ul className="case-list">
			{
				props.caseEntries.map((case_: any, index: number) => (
					<li key={index} onClick={() => props.handleSelectCase(index + (props.offset || 0))}>
						<i>{'[' + case_.date.toLocaleDateString() + '] '}</i>
						{case_.title.substring(0,50) + (case_.title.length > 50 ? '...' : '' )}
					</li>		
				))
			}
			</ul>
		</section>
	)	
}

export default CaseList
