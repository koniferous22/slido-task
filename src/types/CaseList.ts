import { Case } from '../types/Case'

export interface CaseListProps {
	id: string;
	label: string;
	caseEntries: Array<Case>;
	handleSelectCase: (_: number) => void;
	offset?: number;
}
