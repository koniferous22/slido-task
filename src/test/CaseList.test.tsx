'use strict';

import React from 'react'
import renderer from 'react-test-renderer'
import CaseList from '../components/CaseList'

import { Case } from '../types/Case'

it('CaseList renders correctly', () => {
	const date1 = new Date()
	const date2 = new Date()
	date2.setMonth(date1.getMonth() + 1)
	const case1: Case = {
		title: "Dni Zel√° v Stupave",
		description: "this was a truly magnificent event",
		date: date1
	}
	const case2: Case = {
		title: "Filakovsky jarmok",
		description: "Ziedol som 3 a pol porcie segedinskeho gulasu",
		date: date2
	}
  	const tree = renderer
		.create(
			<CaseList
				id="case-list"
				label="Case List"
				caseEntries={[case1, case2]}
				handleSelectCase={(_: number) => {}}
			/>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});