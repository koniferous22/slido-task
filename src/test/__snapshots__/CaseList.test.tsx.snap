// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaseList renders correctly 1`] = `
<section
  id="case-list"
>
  <h4
    className="case-list-title"
  >
    Case List
  </h4>
  <ul
    className="case-list"
  >
    <li
      onClick={[Function]}
    >
      <i>
        [1/25/2020] 
      </i>
      Dni Zelá v Stupave
    </li>
    <li
      onClick={[Function]}
    >
      <i>
        [2/25/2020] 
      </i>
      Filakovsky jarmok
    </li>
  </ul>
</section>
`;

exports[`renders correctly 1`] = `
<section
  id="case-list"
>
  <h4
    className="case-list-title"
  >
    Case List
  </h4>
  <ul
    className="case-list"
  >
    <li
      onClick={[Function]}
    >
      <i>
        [1/25/2020] 
      </i>
      Dni Zelá v Stupave
    </li>
    <li
      onClick={[Function]}
    >
      <i>
        [2/25/2020] 
      </i>
      Filakovsky jarmok
    </li>
  </ul>
</section>
`;
